zio-json/shared/src/main/scala-2.x/zio/json/macros.scala

```diff
@@ -20,8 +20,12 @@ final case class jsonField(name: String) extends Annotation
  */
 final case class jsonAliases(alias: String, aliases: String*) extends Annotation
 
-final class jsonExplicitNull            extends Annotation
-final class jsonExplicitEmptyCollection extends Annotation
+final class jsonExplicitNull extends Annotation
+
+/**
+ * When disabled keys with empty collections will be omitted from the JSON.
+ */
+final case class jsonExplicitEmptyCollection(enabled: Boolean = true) extends Annotation
```

Comment:
There are some binary incompatible changes. And I am not sure we should do this, even if the project is on 0.x
We depend on it in schema and http

Reply:
This was just added in the previous PR and there is no release yet. What else is binary incompatible?

Reply:
The difference between this latest PR and v0.7.3 is a moved JsonCodecConfiguration (was in scala-2 macros.scala and is now in a separate file in shared). 
All other compat issues should now be solved. JsonEncoder does not require the config anymore and is only dropping nulls like in 0.7.3.